package_bag/routines.py

```diff
@@ -0,0 +1,211 @@
+from .models import Bag
+from package_bag.serializers import BagSerializer
+
+from uuid import uuid4
+from shutil import move
+import bagit
+import bagit_profile
+import os
+from os import listdir, rename, remove, mkdir
+from os.path import join, splitext, basename
+from requests import post
+import tarfile
+import json
+
+from zorya import settings
+
+
+class DiscoverBags(object):
+    """
+    Validates bag structure and bag info file, renames bag with unique ID
+    """
+
+    def run(self):
+        processed = []
+        unprocessed = self.discover_bags(settings.SRC_DIR)
+        for u in unprocessed:
+            try:
+                bag_id = self.unpack_rename(u, settings.TMP_DIR)
+                bag = Bag.objects.create(
+                    original_bag_name=u,
+                    bag_identifier=bag_id,
+                    bag_path=join(
+                        settings.TMP_DIR,
+                        bag_id))
+                self.validate_structure(bag.bag_path)
+                self.validate_metadata(bag.bag_path)
+                self.get_data(bag)
+                processed.append(bag)
+            except Exception as e:
+                print(e)
+        # what does this process bags function return? - you want to return something out of the view that indicates which objects were processed
+        return processed
+
+    def discover_bags(self, src):
+        """Looks in a given directory for compressed bags, adds to list to process"""
+        bags_list = []
+        for d in listdir(src):
+            ext = splitext(d)[-1]
+            if ext in ['.tgz', '.tar.gz', '.gz']:
```

Comment:
```suggestion
            if ext in ['.tgz', '.gz']:
```
Pretty sure .tar.gz will never be returned from `splitext` - try it out in an interpreter.